pkg/generic/thrift/write.go

```diff
@@ -54,31 +54,29 @@ var pbGetter fieldGetter = func(val interface{}, field *descriptor.FieldDescript
 
 func typeOf(sample interface{}, t *descriptor.TypeDescriptor, opt *writerOption) (descriptor.Type, writer, error) {
 	tt := t.Type
-	switch v := sample.(type) {
+	switch sample.(type) {
 	case bool:
 		return descriptor.BOOL, writeBool, nil
 	case int8, byte:
-		return descriptor.I08, writeInt8, nil
+		switch tt {
+		case descriptor.I08, descriptor.I16, descriptor.I32, descriptor.I64:
+			return tt, writeInt8, nil
+		}
```

Comment:
what if tt is other than descriptor.I08, descriptor.I16, descriptor.I32, descriptor.I64?

Reply:
the same as line 80 -> 90